### PR TITLE
Add matched groups regardless of existence of unmatched groups

### PR DESCRIPTION
**Pull request by ProBeta1**

### CODE REVIEW

**Review by Brainicism — APPROVED** [review]

File: src/commands/game_options/groups.ts [hunk 1]
```diff
@@ -42,8 +42,7 @@ export default class GroupsCommand implements BaseCommand {
         const { matchedGroups, unmatchedGroups } = await getMatchingGroupNames(groupNames);
         if (unmatchedGroups.length) {
             logger.info(`${getDebugLogHeader(message)} | Attempted to set unknown groups. groups =  ${unmatchedGroups.join(", ")}`);
-            await sendErrorMessage(MessageContext.fromMessage(message), { title: "Unknown Group Name", description: `One or more of the specified group names was not recognized. Please ensure that the group name matches exactly with the list provided by \`${process.env.BOT_PREFIX}help groups\` \nThe following groups were **not** recognized:\n ${unmatchedGroups.join(", ")} ` });
-            return;
+            await sendErrorMessage(MessageContext.fromMessage(message), { title: "Unknown Group Name", description: `One or more of the specified group names was not recognized.Those matched are added. Please ensure that the group name matches exactly with the list provided by \`${process.env.BOT_PREFIX}help groups\` \nThe following groups were **not** recognized:\n ${unmatchedGroups.join(", ")} ` });
         }
         guildPreference.setGroups(matchedGroups);
```
Comment (Brainicism) [inline src/commands/game_options/groups.ts:47]: Make sure there is atleast one matched group before `setGroups`

Comment (Brainicism) [reply src/commands/game_options/groups.ts:47]: Make the same change in `add`, `exclude`, include`, and `remove`

**Review by Brainicism — CHANGES_REQUESTED** [review]

File: src/commands/game_options/add.ts [hunk 2]
```diff
@@ -71,27 +71,28 @@ export default class AddCommand implements BaseCommand {
...
-                await sendOptionsMessage(MessageContext.fromMessage(message), guildPreference, { option: GameOption.INCLUDE, reset: false });
-                logger.info(`${getDebugLogHeader(message)} | Include added: ${guildPreference.getDisplayedIncludesGroupNames()}`);
-                break;
-            case AddType.EXCLUDES:
-                guildPreference.setExcludes(matchedGroups);
-                await sendOptionsMessage(MessageContext.fromMessage(message), guildPreference, { option: GameOption.EXCLUDE, reset: false });
-                logger.info(`${getDebugLogHeader(message)} | Exclude added: ${guildPreference.getDisplayedExcludesGroupNames()}`);
-                break;
-            default:
+        if (matchedGroups.length) {
```
Comment (Brainicism) [inline src/commands/game_options/add.ts:77]: Do an early `return` if `matchedGroups` is empty to avoid nesting. Same with the rest of the changes

**Review by Brainicism — CHANGES_REQUESTED** [review]

File: src/commands/game_options/add.ts (hunk 2 above)
Comment (ProBeta1) [reply src/commands/game_options/add.ts:77]: okay

**Review by Brainicism — APPROVED** [review]

**Review by Brainicism — DISMISSED** [review]

**Review by Brainicism — APPROVED** [review]

**Pull request merged**